File: oneswiss-server/Models/EventLogSettings.cs
using System.ComponentModel.DataAnnotations;

namespace OneSwiss.Server.Models;

public class EventLogSettings : DatabaseObject
{
    public bool Enabled { get; set; }

    public Guid? DbmsId { get; set; }

    
    [MaxLength(100)]
    public string DatabaseName { get; set; } = string.Empty;

    
    [MaxLength(100)]
    public string Table { get; set; } = string.Empty;

    
    public Guid? CredentialsId { get; set; }
    
    
    public string InfoBaseNameRegex { get; set; } = string.Empty;
    public int DefaultTtl { get; set; } = 365;

    public Dbms? Dbms { get; set; }
    public Credentials? Credentials { get; set; }
}

File: oneswiss-web/src/lib/api/eventlog-settings.ts
import { apiGet, apiPut } from "@/lib/api/client";

export type EventLogLookupItem = {
  id: string;
  name: string;
};

export type EventLogExportItem = {
  infoBaseId: string;
  isActive: boolean;
  ttl: number;
  reduceSourceLog: boolean;
  reduceKeepDays: number;
  lastReducedUpTo: string | null;
};

export type EventLogSettingsItem = {
  enabled: boolean;
  dbmsId: string | null;
  databaseName: string;
  table: string;
  credentialsId: string | null;
  infoBaseNameRegex: string;
  defaultTtl: number;
  reductionEnabled: boolean;
  reductionHourUtc: number;
  reductionSafetyMarginHours: number;
};

export type EventLogSettingsResponse = {
  settings: EventLogSettingsItem;
  items: EventLogExportItem[];
  dbms: EventLogLookupItem[];
  credentials: EventLogLookupItem[];
  infoBases: EventLogLookupItem[];
};

export type SaveEventLogSettingsRequest = {
  enabled: boolean;
  dbmsId: string | null;
  databaseName: string;
  table: string;
  credentialsId: string | null;
  infoBaseNameRegex: string;
  defaultTtl: number;
  reductionEnabled: boolean;
  reductionHourUtc: number;
  reductionSafetyMarginHours: number;
  items: EventLogExportItem[];
};

export function getEventLogSettings() {
  return apiGet<EventLogSettingsResponse>("/api/eventlog/settings");
}

export function saveEventLogSettings(request: SaveEventLogSettingsRequest) {
  return apiPut<EventLogSettingsItem, SaveEventLogSettingsRequest>("/api/eventlog/settings", request);
}
